IntegrationTests/tests_04_performance/test_01_resources/test_1000_udpconnections.swift

```diff
@@ -0,0 +1,56 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2020 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+fileprivate final class DoNothingHandler: ChannelInboundHandler {
+    public typealias InboundIn = ByteBuffer
+    public typealias OutboundOut = ByteBuffer
+}
+
+func run(identifier: String) {
+    let serverChannel = try! DatagramBootstrap(group: group)
+        .channelOption(ChannelOptions.socketOption(.so_reuseaddr), value: 1)
+        // Set the handlers that are applied to the bound channel
+        .channelInitializer { channel in
+            return channel.pipeline.addHandler(DoNothingHandler())
+        }
+        .bind(to: localhostPickPort).wait()
+    defer {
+        try! serverChannel.close().wait()
+    }
+
+    let remoteAddress = serverChannel.localAddress!
+    
+    let clientBootstrap = DatagramBootstrap(group: group)
+            .channelOption(ChannelOptions.socketOption(.so_reuseaddr), value: 1)
+
+    measure(identifier: identifier) {
+        let numberOfIterations = 1000
+        for _ in 0 ..< numberOfIterations {
+            let clientChannel = try! clientBootstrap.bind(to: localhostPickPort).wait()
+            defer {
+                try! clientChannel.close().wait()
+            }
+            
+            // Send a byte to make sure everything is really open.
+            var buffer = clientChannel.allocator.buffer(capacity: 1)
```

Comment:
I still think we should just move these two lines out. `clientChannel.allocator.buffer(capacity: 1)` has to allocate twice (the `class` and the bytes) and we don't need to change this buffer.